discovery-infra/bootstrap_in_place.py

```diff
@@ -0,0 +1,171 @@
+
+
+def embed(image_name, ignition_file, embed_image_name):
+    logging.info("Embed ignition %s to iso %s", ignition_file, image_name)
+    embedded_image = os.path.join(BUILD_DIR, embed_image_name)
+    os.remove(embedded_image) if os.path.exists(embedded_image) else None
+
+    flags = shlex.split(f"--privileged --rm -v /dev:/dev -v /run/udev:/run/udev -v .:/data -w /data")
+    utils.run_container("coreos-installer", "quay.io/coreos/coreos-installer:release", flags,
+                        f"iso ignition embed {BUILD_DIR}/{image_name} "
+                        f"-f --ignition-file /data/{IBIP_DIR}/{ignition_file} -o /data/{embedded_image}")
+
+    image_path = os.path.join(consts.BASE_IMAGE_FOLDER, embed_image_name)
+    shutil.move(embedded_image, image_path)
+    return image_path
+
+
+def fill_install_config(pull_secret, ssh_pub_key, net_asset, cluster_name):
+    yaml.add_representer(str, str_presenter)
+    with open(INSTALL_CONFIG, "r") as _file:
+        config = yaml.safe_load(_file)
+
+    config["pullSecret"] = pull_secret
+    config["sshKey"] = ssh_pub_key
+    config["metadata"]["name"] = cluster_name
+    config["networking"]["machineNetwork"][0]["cidr"] = net_asset["machine_cidr"]
+
+    with open(INSTALL_CONFIG, "w") as _file:
+        yaml.dump(config, _file)
+
+
+def setup_files_and_folders(args, net_asset, cluster_name):
+    logging.info("Creating needed files and folders")
+    utils.recreate_folder(consts.BASE_IMAGE_FOLDER, force_recreate=False)
+    utils.recreate_folder(IBIP_DIR, with_chmod=False, force_recreate=True)
+    shutil.copy(os.path.join(RESOURCES_DIR, INSTALL_CONFIG_FILE_NAME), IBIP_DIR)
+    fill_install_config(args.pull_secret, args.ssh_key, net_asset, cluster_name)
+
+
+def str_presenter(dumper, data):
+    if "ssh-rsa" in data:  # check for multiline string
+        return dumper.represent_scalar('tag:yaml.org,2002:str', data, style='|')
+    return dumper.represent_scalar('tag:yaml.org,2002:str', data)
+
+
+def create_controller(net_asset):
+    return TerraformController(
+        cluster_name="test-infra-cluster",
+        num_masters=1,
+        num_workers=0,
+        master_memory=32 * 1024,  # 32GB of RAM
+        master_vcpu=12,
+        net_asset=net_asset,
+        iso_download_path="<TBD>",  # will be set later on
+        bootstrap_in_place=True,
+    )
+
+
+def all_operators_up():
+    statuses = get_operators_status(KUBE_CONFIG)
+    if not statuses:
+        logging.debug("No operator has been found currently...")
+        return False
+
+    invalid_operators = [operator for operator, up in statuses.items() if not up]
+
+    all_operators_are_valid = len(invalid_operators) == 0
+
+    if not all_operators_are_valid:
+        logging.debug("Following operators are still down: %s", ", ".join(invalid_operators))
+
+    return all_operators_are_valid
+
+
+def gather_sosreport_data(host_ip):
+    with SshConnection(ip=host_ip, private_ssh_key_path=SSH_KEY) as ssh:
```

Comment:
we have this option in Node object. Why not to use the object?

Reply:
and add function that will upload file will be very helpful to there

Reply:
ok